src/error.ts

```diff
@@ -1019,8 +1022,10 @@ export class MongoMissingDependencyError extends MongoAPIError {
    *
    * @public
    **/
-  constructor(message: string, options: { cause?: Error } = {}) {
+  constructor(message: string, options: { cause: Error; dependencyName: string }) {
     super(message, options);
+    this.dependencyName = options.dependencyName;
+    this.cause = options.cause;
```

Comment:
Why did we add this? can't we just let cause travel up to the super class?

Reply:
Because I `override cause: Error` in this class (it's now required to construct a MissingDependencyError), TS complains that we didn't initialize the `cause` property in the constructor even though it's assigned in the superclass.

Reply:
I assume you added `override cause: Error` because it will type `cause` to 1. always exist and 2. be an error type rather than (I think) `unknown`? 

Could we use `!` like: `override cause!: Error`? By writing `this.cause =` this "cause" property has a different descriptor than if we allow the error super class to attach it.

Reply:
(maybe got lost) can we make this change?